### PR TITLE
test(other): migrate cypress to v12

### LINKED ISSUE #6118: 💥 [DevOps] migrate Cypress to current version 12
## 💥 DevOps ticket
For end-to-end testing Cypress version 7 is in use.
We should migrate to the current version 12.

### TODO
- [x] upgrade Cypress version 
- [x] migrate cucumber preprocessor to [@badeball/cypress-cucumber-preprocessor](https://github.com/badeball/cypress-cucumber-preprocessor)
- [x] migrate configuration JSON to Javascript
- [x] change test directory structure from `integration` to `e2e` 
- [x] fix Cypress issue with finding model definitions for factories

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest
Upgrade the Cypress framework to v12

### Issues
- fixes #6118

### Todo
- [x] fix Cypress issue with finding model definitions for factories

### CODE REVIEW

File: cypress/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (results) {
+      console.log(results.status);
+    }
+  });
+
+  return config;
+}
+
+// Import backend .env (smart)?
+const { parsed } = dotenv.config({ path: '../backend/.env' })
```
Comment (mahula) [inline cypress/cypress.config.js:35]: Research, if there is a better practice for Cypress wil be covered in issue  #6374.

File: cypress/support/step_definitions/Moderation.ReportContent/somebody_reported_the_following_posts.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { Given } from "cypress-cucumber-preprocessor/steps";
+import { Given } from "@badeball/cypress-cucumber-preprocessor";
+import { gql } from '../../../../backend/src/helpers/jest'
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply cypress/support/step_definitions/Moderation.ReportContent/somebody_reported_the_following_posts.js:2]:
```suggestion
```

**Review by Elweyn — CHANGES_REQUESTED** [review]
Works for me!
Just the import of gql can be removed!

**Comment by ulfgebhardt** [comment]
Try updating `neode` in `root/package.json ` -> to avoid: 
```
Image: typeof Cypress !== 'undefined' ? require('./Image.js') : require('./Image.js').default,
```

**Comment by mahula** [comment]
> Try updating `neode` in `root/package.json ` -> to avoid:
> 
> ```
> Image: typeof Cypress !== 'undefined' ? require('./Image.js') : require('./Image.js').default,
> ```

Does neode v0.4.9 really solve this?

**Comment by Tirokk** [comment]
I just tried it out to see if it runs on my machine and get this error:

Bildschirmfoto 2023-06-06 um 13 39 20

**Review by Elweyn — APPROVED** [review]

**Pull request merged**